lib/rules/no-double-assignments.js

```diff
@@ -0,0 +1,81 @@
+"use strict";
+
+function getPrevNode(context, node) {
+    var prevToken = context.getTokenBefore(node);
+    if (prevToken) {
+        return context.getNodeByRangeIndex(prevToken.range[0]);
+    }
+}
+
+function expressionname(expression) {
+    if (expression.type === "Identifier") {
+        return expression.name;
+    }
+
+    if (expression.type === "MemberExpression") {
+        return expressionname(expression.object) + "-" + expression.property.name;
+    }
+}
+
+function varname(node) {
+    if (!node || node.type !== "ExpressionStatement") {
+        return null;
+    }
+
+    if (node.expression.type !== "AssignmentExpression") {
+        return null;
+    }
+
+    return expressionname(node.expression.left);
+}
+
+function any(list, testCb) {
+    for (var i = list.length - 1; i >= 0; i--) {
+        if (testCb(list[i])) {
+            return true;
+        }
+    }
+    return false;
+}
+
+function includesname(name, expression) {
```

Comment:
Our convention is to use camelcase for all identifers, and function names must have verbs. Please see: http://eslint.org/docs/developer-guide/code-conventions.html